src/app/core/entity/entity-actions/entity-edit.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from "@angular/core";
+import { EntityMapperService } from "../entity-mapper/entity-mapper.service";
+import { Entity } from "../model/entity";
+import { EntitySchemaService } from "../schema/entity-schema.service";
+import { CascadingEntityAction } from "./cascading-entity-action";
+import { UnsavedChangesService } from "app/core/entity-details/form/unsaved-changes.service";
+
+/**
+ * Safely edit an entity including handling references with related entities.
+ * This service is usually used in combination with the `EntityActionsService`, which provides user confirmation processes around this.
+ */
+@Injectable({
+  providedIn: "root",
+})
+export class EntityEditService extends CascadingEntityAction {
+  constructor(
+    protected override entityMapper: EntityMapperService,
+    protected override schemaService: EntitySchemaService,
+    private unsavedChanges: UnsavedChangesService,
+  ) {
+    super(entityMapper, schemaService);
+  }
+
+  async editEntity<E extends Entity>(
+    result: any,
```

Comment:
what is "result" supposed to contain? as it is an input parameter, this name doesn't seem very well fitting?

Can you define the type of this also?

Reply:
@sleidig I have renamed "result" to "updatedEntity" to reflect its purpose better. Additionally, since the label can hold various types of values (like a string or an array), I updated the type definition to:`updatedEntity: {selectedField: string, label: E| E[]},`
Please let me know if this approach makes sense or if you have further suggestions.

Reply:
did some changes - 
1.  rename the label.
2. changed the type to "any"